core/tests/widgets/test_base.py

```diff
@@ -277,12 +277,15 @@ def test_add_multiple_children(app, widget):
     assert_action_performed_with(window.content, "refresh")
 
     # App's widget index has been updated
-    assert dict(app.widgets) == {
-        "widget_id": widget,
-        "child1_id": child1,
-        "child2_id": child2,
-        "child3_id": child3,
-    }
+    assert (
+        dict(app.widgets).items()
+        >= {
+            "widget_id": widget,
+            "child1_id": child1,
+            "child2_id": child2,
+            "child3_id": child3,
+        }.items()
```

Comment:
While this does pass, It's not a very robust test.

`dict.items()` will be converting the dictionary into a list of tuples; the list comparison will be trying to evaluate that the two lists sort in lexical order. In effect, this test asserts that the `app.widgets` list has, at the bare minimum, a single widget that is alphabetically sorted before "widget_id". This will evaluate as true in this case... but so would *any* app widget list that contains at least one item (`{'foo': widget}` would pass this test, for example, even though it doesn't contain any of the child widgets).

If the intention is to validate that "all of these objects in app.widgets", it would be better to evaluate each membership specifically; or, alternatively, use a subset-based check (possibly factored out as a standalone assertion).

Another approach in this case would be even simpler - use a reference to the literal "extra" widget that is causing an issue for the test. The extra widget that prevents the use of a literal `==` here is the content widget of the test app's main window - something that isn't involved in this test, but *is* in the app registry. Adding `app.main_window.content.id: app.main_window.content,` as an additional pair in the expected output is a reasonable approach in this case - we don't care about the specific ID or widget; we just care that the app's widget registry contains a reference to all known widgets - and the app's (default) content is one of those widgets.